### PR TITLE
Migrate ReportConfig to sql

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Summary

Changes to migrate ReportConfig from couch to SQL
## Safety Assurance

- [ ] Risk label is set correctly
- [ ] All migrations are backwards compatible and won't block deploy
- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [ ] If QA is part of the safety story, the "Awaiting QA" label is used
- [ ] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by AmitPhulera** [comment]
@millerdev, @dannyroberts, @orangejenny I followed the documentation and was able to raise this PR. 
Looking for some early suggestions on the PR.

File: corehq/apps/saved_reports/models.py [hunk 1]
```diff
@@ -792,3 +829,98 @@ def get_latest(cls):
             return cls.objects.order_by('-end_datetime')[0]
         except IndexError:
             return None
+
+class SQLReportConfig(SyncSQLToCouchMixin, models.Model):
+    domain = models.CharField(max_length=32, null=False, db_index=True)
```
Comment (snopoke) [inline corehq/apps/saved_reports/models.py:834]: I think this max length has to be longer, at least 128 chars

File: corehq/apps/saved_reports/models.py [hunk 2]
```diff
@@ -792,3 +829,98 @@ def get_latest(cls):
...
+    # unknown fields
+    update_seq = models.CharField(max_length=1000, null=False)
+    purge_seq = models.IntegerField(null=False)
+    db_name = models.CharField(max_length=12, null=False)
+    doc_del_count = models.IntegerField(null=False)
+    instance_start_time = models.CharField(max_length=1, null=False)
+    disk_size = models.IntegerField(null=False)
+    doc_count = models.IntegerField(null=False)
+    disk_format_version = models.IntegerField(null=False)
+    data_size = models.IntegerField(null=False)
```
Comment (snopoke) [inline corehq/apps/saved_reports/models.py:857]: are these necessary?

File: corehq/apps/saved_reports/models.py [hunk 3]
```diff
@@ -792,3 +829,98 @@ def get_latest(cls):
...
+            "disk_format_version",
+            "data_size",
+        ]
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return ReportConfig
+
+
+# TODO: Add SyncCouchToSQLMixin and the following methods to ReportConfig
```
Comment (snopoke) [inline corehq/apps/saved_reports/models.py]: unnecessary comment

File: corehq/apps/saved_reports/models.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/saved_reports/models.py:834]: Also wondering about the length of some of the other fields

File: corehq/apps/saved_reports/models.py (hunk 3 above)
Comment (AmitPhulera) [reply corehq/apps/saved_reports/models.py]: Removed.

File: corehq/apps/saved_reports/models.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/saved_reports/models.py:857]: I was not sure about what purpose do these fields would be serving. I found some references of the fields in some pillow files but not entirely sure if they belong to ReportConfig. So I kept them with a comment.

File: corehq/apps/saved_reports/models.py (hunk 1 above)
Comment (AmitPhulera) [reply corehq/apps/saved_reports/models.py:834]: This was the default code generated by `evaluate_couch_model_for_sql`. I still have to fine-tune these variables.

Comment (orangejenny) [reply corehq/apps/saved_reports/models.py:834]: Domain should probably be treated as a special case in `evaluate_couch_model_for_sql`, I had thought it already was but see it isn't. We limit domain names in the UI to 25 chars, so it's only domains that are 5+ years old that would have longer names, but they do exist.

File: corehq/apps/saved_reports/models.py [hunk 4]
```diff
@@ -792,3 +828,65 @@ def get_latest(cls):
...
+
+
+class SQLReportConfig(SyncSQLToCouchMixin, models.Model):
+    domain = models.CharField(max_length=32, null=False, db_index=True)
+    report_type = models.CharField(max_length=32, null=False)
+    report_slug = models.CharField(max_length=32, null=False)
+    subreport_slug = models.CharField(max_length=64, null=True)
+    name = models.CharField(max_length=128, null=False)
+    description = models.CharField(max_length=1000, null=True)
+    owner_id = models.CharField(max_length=64, null=False)
```
Comment (orangejenny) [inline corehq/apps/saved_reports/models.py:840]: Curious that this came out as 64 - isn't this a couch id?

File: corehq/apps/saved_reports/models.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/saved_reports/models.py:857]: I think these should be deleted if there are no meaningful code references to them.

**Pull request closed without merging**